File: games/mafia/backend/app/schemas/game_info.py
from pydantic import BaseModel


class RoleInfoOut(BaseModel):
    key: str
    display_name: str
    team: str
    description: str
    example: str
    acts_at_night: bool
    allow_self_target: bool
    hostile: bool
    max_uses: int | None
    mutually_exclusive_with: list[str]


class PhaseInfoOut(BaseModel):
    key: str
    name: str
    summary: str
    details: str
    example: str


class TieBreakerOptionOut(BaseModel):
    key: str
    label: str
    description: str
    example: str


class TieBreakerInfoOut(BaseModel):
    key: str
    title: str
    description: str
    options: list[TieBreakerOptionOut]


class RuleSectionOut(BaseModel):
    title: str
    body: str
    example: str | None = None


class FaqEntryOut(BaseModel):
    question: str
    answer: str


class GameInfoOut(BaseModel):
    """The full reference-data payload for the Roles / Gameplay /
    Tie-Breakers / Rules / FAQ pages — one endpoint, since it's small and
    entirely static, rather than a round trip per page.
    """

    roles: list[RoleInfoOut]
    phases: list[PhaseInfoOut]
    tie_breakers: list[TieBreakerInfoOut]
    rules: list[RuleSectionOut]
    faq: list[FaqEntryOut]

File: games/mafia/frontend/types/game-info.ts
export interface RoleInfo {
  key: string;
  display_name: string;
  team: string;
  description: string;
  example: string;
  acts_at_night: boolean;
  allow_self_target: boolean;
  hostile: boolean;
  max_uses: number | null;
  mutually_exclusive_with: string[];
}

export interface PhaseInfo {
  key: string;
  name: string;
  summary: string;
  details: string;
  example: string;
}

export interface TieBreakerOption {
  key: string;
  label: string;
  description: string;
  example: string;
}

export interface TieBreakerInfo {
  key: string;
  title: string;
  description: string;
  options: TieBreakerOption[];
}

export interface RuleSection {
  title: string;
  body: string;
  example: string | null;
}

export interface FaqEntry {
  question: string;
  answer: string;
}

export interface GameInfo {
  roles: RoleInfo[];
  phases: PhaseInfo[];
  tie_breakers: TieBreakerInfo[];
  rules: RuleSection[];
  faq: FaqEntry[];
}
